### PR TITLE
Terraform module and documentation

### PR DESCRIPTION
**Pull request by andybkay**
I's finally here, sorry it took so long I've been trying to wrap up a project. Fully tested with multiple versions of terraform. Any questions, please let me know.

I've also updated the original README_DEPLOY.md file to include a link to the readme inside the terraform directory.

### CODE REVIEW

**Comment by andybkay** [comment]
There was a merge conflict in the docs, fixed

**Comment by alexcasalboni** [comment]
Thank you, @andybkay 🙏 

This looks awesome 🚀 

I've only left a comment about locals reuse (not a Terraform expert here, sorry). Does it make sense to you?

File: terraform/module/lambda.tf [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  # source_code_hash = "${base64sha256(file("lambda_function_payload.zip"))}"
+  source_code_hash = data.archive_file.app.output_base64sha256
+
+  runtime = "nodejs14.x"
+
+  environment {
+    variables = {
+      defaultPowerValues = "[128,256,512,1024,1536,3008]"
+      minRAM             = 128
+      baseCosts          = jsonencode({ "ap-east-1" : 2.9e-9, "af-south-1" : 2.8e-9, "me-south-1" : 2.6e-9, "eu-south-1" : 2.4e-9, "ap-northeast-3" : 2.7e-9, "default" : 2.1e-9 }),
```
[the remark this answers is not in the document]
Comment (andybkay) [reply terraform/module/lambda.tf:22]: This was the original plan, hence the locals. Looks like I missed this out. I'll drop a fix in now

**Comment by andybkay** [comment]
So, it looks like the script 'build-layer.sh' has been removed from the repo. Do you know what it's been replaced with @alexcasalboni

**Comment by andybkay** [comment]
> I've only left a comment about locals reuse (not a Terraform expert here, sorry). Does it make sense to you?

It's something I had planned on, but hadn't integrated fully into the lambda.tf file. I've put the change in locally while I figure out the build-layer.sh thing

**Comment by andybkay** [comment]
Ok, final commit. Looks like there were quite a few changes between when I first wrote this and today, but those tweaks have now been integrated and overcome. This is working now with all versions mentioned.

**Comment by alexcasalboni** [comment]
Thank you @andybkay, looks great!

I've applied a couple of minor edits in the new docs to use a tone similar to the rest of the docs.

It looks like we're ready to merge now :) Can I thank you officially somewhere on Twitter or LinkedIn?

**Comment by andybkay** [comment]
Oh wow that'd be awesome: [LinkedIn](https://www.linkedin.com/in/andrew-kay-a5a833/)

Thanks

**Comment by alexcasalboni** [comment]
Apologies, it took me a while to merge & thank you :)

**Pull request merged**